### PR TITLE
[TH2-4566] reusable workflow

### PR DESCRIPTION
**Pull request by ak49630**
[TH2-4566] reusable workflow

[TH2-4566]: https://exactpro.atlassian.net/browse/TH2-4566?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -103,8 +102,9 @@ dependencies {
     api platform('com.exactpro.th2:bom:4.1.0')
 
     implementation "com.exactpro.th2:common:${commonVersion}"
-    implementation 'com.exactpro.th2:codec:4.7.4'
-    implementation 'com.exactpro.th2:sailfish-utils:3.13.0'
+    implementation 'com.exactpro.th2:codec:4.7.5'
+    implementation 'com.exactpro.th2:sailfish-utils:3.14.0'
```
Comment (Nikita-Smirnov-Exactpro) [inline build.gradle:106]: Try to migrate to sailfish-common

File: build.gradle [hunk 2]
```diff
@@ -151,7 +151,3 @@ compileTestKotlin {
         jvmTarget = "11"
     }
 }
-
```
Comment (Nikita-Smirnov-Exactpro) [inline build.gradle:154]: If you migrate to AutoService than please add

apply plugin: 'kotlin-kapt'

dependencies {
    annotationProcessor 'com.google.auto.service:auto-service:1.0.1'
    kapt 'com.google.auto.service:auto-service:1.0.1'
}

File: build.gradle [hunk 3]
```diff
@@ -103,8 +102,9 @@ dependencies {
     api platform('com.exactpro.th2:bom:4.1.0')
 
     implementation "com.exactpro.th2:common:${commonVersion}"
-    implementation 'com.exactpro.th2:codec:4.7.4'
-    implementation 'com.exactpro.th2:sailfish-utils:3.13.0'
+    implementation 'com.exactpro.th2:codec:4.7.5'
+    implementation 'com.exactpro.th2:sailfish-utils:3.14.0'
+    implementation 'com.google.auto.service:auto-service:1.0.1'
```
Comment (Nikita-Smirnov-Exactpro) [inline build.gradle:107]:
```suggestion
    compileOnly 'com.google.auto.service:auto-service:1.0.1'
```

**Review by Nikita-Smirnov-Exactpro — CHANGES_REQUESTED** [review]

File: build.gradle (hunk 1 above)
Comment (isengrims) [reply build.gradle:107]: Is there IMessage to Proto conversion in sailfish-common? Seems like there is no such functionality in common

**Review by Nikita-Smirnov-Exactpro — APPROVED** [review]

**Pull request merged**